### PR TITLE
implement api_addr parameter

### PR DESCRIPTION
**Pull request by bastelfreak**
##### SUMMARY

This pull request adds support for the api_addr parameter. It's upstream documented at: https://www.vaultproject.io/docs/configuration/index.html#api_addr

##### TESTS/SPECS

I updated the unit tests

The tests work and I already use this in production

### CODE REVIEW

**Comment by bastelfreak** [comment]
@jsok ping :)

File: spec/classes/vault_spec.rb [hunk 1]
```diff
@@ -125,6 +126,21 @@
           }
         end
 
+        context 'when disable mlock' do
```
Comment (jsok) [inline spec/classes/vault_spec.rb]: please update to match the test intention

File: manifests/init.pp [hunk 2]
```diff
@@ -61,6 +61,13 @@
 #   because Vault can block a scheduler thread". Default: number of CPUs
 #   on the system, retrieved from the ``processorcount`` Fact.
 #
+# * `api_addr`
```
Comment (jsok) [inline manifests/init.pp:64]: Could you please document this in the top level `README.me` as well?

**Review by jsok — CHANGES_REQUESTED** [review]

File: spec/classes/vault_spec.rb (hunk 1 above)
Comment (bastelfreak) [reply spec/classes/vault_spec.rb]: done

File: manifests/init.pp (hunk 2 above)
Comment (bastelfreak) [reply manifests/init.pp:64]: done

**Comment by bastelfreak** [comment]
@jsok ping :)

**Review by jsok — APPROVED** [review]
Thanks  👍 
the rubocop stuff is a real pain to keep up to date :(

**Pull request merged**